### PR TITLE
[FLINK-7053][blob] improve code quality in some tests

### PR DESCRIPTION
**Pull request by NicoK**
* `BlobClientTest` and `BlobClientSslTest` share a lot of common code
* the received buffers there are currently not verified for being equal to the expected one
* `TemporaryFolder` should be used throughout blob store tests

This PR is based upon #4158 in a series of PRs to implement FLINK-6916.

### CODE REVIEW

**Comment by zentol** [comment]
@NicoK Could you rebase this PR?

**Comment by NicoK** [comment]
ok, I rebased this and all following PRs for FLINK-6916

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobCacheSuccessTest.java [hunk 1]
```diff
@@ -92,15 +100,15 @@ private void uploadFileGetTest(final Configuration config, boolean cacheWorksWit
...
-				// just in case parameters are still read from the server,
-				// create a separate configuration object for the cache
-				cacheConfig = new Configuration(config);
+			final Configuration cacheConfig = new Configuration(config);
+			cacheConfig.setString(BlobServerOptions.STORAGE_DIRECTORY,
+				temporaryFolder.newFolder().getAbsolutePath());
+			if (!cacheHasAccessToFs) {
+				// make sure the cache cannot access the HA store directly
+				cacheConfig.setString(BlobServerOptions.STORAGE_DIRECTORY,
+					temporaryFolder.newFolder().getAbsolutePath());
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobCacheSuccessTest.java]: isn't this redundant?

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java [hunk 2]
```diff
@@ -107,195 +96,89 @@ public static void stopServers() throws IOException {
 		if (BLOB_SSL_SERVER != null) {
 			BLOB_SSL_SERVER.close();
 		}
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java]: What about shutting the non ssl blob server down?

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java [hunk 3]
```diff
@@ -107,195 +96,89 @@ public static void stopServers() throws IOException {
...
 	@Test
-	public void testSSLClientFailure() throws Exception {
-		try {
-			uploadJarFile(BLOB_SERVER, sslClientConfig);
-			fail("SSL client connected to non-ssl server");
-		} catch (Exception e) {
-			// Exception expected
-		}
+	public void testNonSSLConnection2() throws Exception {
+		uploadJarFile(BLOB_SERVER, nonSslClientConfig);
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java]: Why are we testing BLOB_SERVER here? Isn't this part of the `BlobClientTest`?

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobCacheSuccessTest.java (hunk 1 above)
Comment (NicoK) [inline flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobCacheSuccessTest.java]: seems that it wasn't used to its full extension - also the meaning of this variable was not what clear

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java (hunk 2 above)
Comment (NicoK) [inline flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java]: good catch

File: flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java (hunk 3 above)
Comment (NicoK) [inline flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobClientSslTest.java]: the client configuration is different - although both have `BlobServerOptions.SSL_ENABLED` disabled, this one has `SecurityOptions.SSL_ENABLED` enabled

**Comment by tillrohrmann** [comment]
Changes look good to me. Thanks for the work @NicoK. Merging this PR.

**Pull request closed without merging**